### ISSUE TITLE
akka-stream-kafka Cannot push port twice, or before it being pulled

### ISSUE THREAD

**Issue by Zhen-hao** [akka/alpakka-kafka#1165]
Hi, 

I encountered this elusive bug on production and don't have a good way to reproduce it. 

With Akka 2.6.6 and `akka-stream-kafka` 2.0.3, I got the following error.

```
j.l.IllegalArgumentException: Cannot push port (FlowOut(1694942328)) twice, or before it being pulled
	at a.stream.stage.GraphStageLogic.push(GraphStage.scala:635)
	at a.k.i.CommitCollectorStageLogic.$anonfun$onTimer$1(CommitCollectorStage.scala:68)
	at a.k.i.CommitCollectorStageLogic.$anonfun$onTimer$1$adapted(CommitCollectorStage.scala:68)
	at a.k.i.CommitCollectorStageLogic.akka$kafka$internal$CommitCollectorStageLogic$$pushDownStream(CommitCollectorStage.scala:76)
	at a.k.i.CommitCollectorStageLogic.onTimer(CommitCollectorStage.scala:68)
	at a.s.stage.TimerGraphStageLogic.onInternalTimer(GraphStage.scala:1602)
	at a.s.stage.TimerGraphStageLogic.$anonfun$getTimerAsyncCallback$1(GraphStage.scala:1591)
	at a.s.stage.TimerGraphStageLogic.$anonfun$getTimerAsyncCallback$1$adapted(GraphStage.scala:1591)
	at a.s.i.fusing.GraphInterpreter.runAsyncInput(GraphInterpreter.scala:466)
	at a.s.i.f.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:498)
	at a.s.i.f.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:600)
	at a.s.i.f.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:769)
	at a.s.i.f.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:784)
	at akka.actor.Actor.aroundReceive(Actor.scala:535)
	at akka.actor.Actor.aroundReceive$(Actor.scala:533)
	at a.s.i.f.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:691)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:575)
	at akka.actor.ActorCell.invoke(ActorCell.scala:545)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
	at akka.dispatch.Mailbox.run(Mailbox.scala:231)
	... 5 frames truncated
```

I solved it by downgrading to Akka 2.5.31 and `akka-stream-kafka` 1.1.0.

**Comment by seglo** [comment]
Hi @Zhen-hao. This issue was just fixed (https://github.com/akka/alpakka-kafka/issues/1159) and we released Alpakka Kafka 2.0.4 yesterday.  Let me know if you still encounter issues with the latest version.

**Comment by Zhen-hao** [comment]
@seglo thank you for the update. I will try it soon.
